### PR TITLE
add sccache recipe

### PR DESCRIPTION
**Pull request by luizirber**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/sccache/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - rust
+    - {{ compiler('c') }}              # [not win]
+    - {{ compiler('m2w64_c') }}        # [win]
+    - openssl
```
Comment (marcelotrevisani) [inline recipes/sccache/meta.yaml]: ``openssl`` should not be here, could you please delete it?

File: recipes/sccache/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+test:
+  commands:
+    - sccache --help
+
+about:
+  home: https://github.com/mozilla/sccache
+  summary: 'sccache is ccache with cloud storage'
+  license: Apache 2.0
```
Comment (marcelotrevisani) [inline recipes/sccache/meta.yaml]:
```suggestion
  license: Apache-2.0
```

File: recipes/sccache/meta.yaml (hunk 1 above)
Comment (marcelotrevisani) [reply recipes/sccache/meta.yaml]: I think ``rust`` should be in ``host`` and ``run`` requirements, am I right?

**Comment by luizirber** [comment]
thanks @marcelotrevisani! Seems like I'm overlinking [`libresolv.9`][0] on macOS, pointers on how to fix it?

[0]: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=107182&view=logs&j=c572e5d5-7ba7-55ce-5575-7ce1d9f7161e&t=81967b5f-0283-5c0b-87c1-e8fdeef842a3&l=587

File: recipes/sccache/meta.yaml (hunk 1 above)
Comment (luizirber) [reply recipes/sccache/meta.yaml]: You don't need `rust` to run, it generates a static binary (but linking to openssl dinamically). Is `build` requirements only for `{{ compiler }}` deps?

**Comment by luizirber** [comment]
should I whitelist `libresolv.9.dylib`, like the [`cyrus-sasl`][0] feedstock does?

[0]: https://github.com/conda-forge/cyrus-sasl-feedstock/blob/9a84fd3bdf037c891971f374813d1829eab86cbd/recipe/meta.yaml#L27L29

File: recipes/sccache/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+build:
+  number: 0
+  missing_dso_whitelist:          # [osx]
+    - /usr/lib/libresolv.9.dylib  # [osx]
+
+requirements:
+  build:
+    - rust
+    - {{ compiler('c') }}              # [not win]
+    - {{ compiler('m2w64_c') }}        # [win]
```
Comment (luizirber) [inline recipes/sccache/meta.yaml]: what is the proper way to choose a compiler for windows? I used the `ripgrep` feedstock as example, and it was using this one.

**Comment by luizirber** [comment]
Ready for review @conda-forge/staged-recipes

File: recipes/sccache/meta.yaml (hunk 3 above)
Comment (mariusvniekerk) [reply recipes/sccache/meta.yaml]: This looks fine. If your package can build with msvc instead of the gnu ones you can just use compiler('c')

Comment (luizirber) [reply recipes/sccache/meta.yaml]: I tried with `compiler('c')`, but then it couldn't link openssl. Reverting to `compiler('m2w64_c')` for now.

**Pull request merged**